types/src/transaction/script.rs

```diff
@@ -56,7 +56,27 @@ impl fmt::Debug for Script {
 
 /// How to call a particular Move script (aka. an "ABI").
 #[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
-pub struct ScriptABI {
+pub enum ScriptABI {
+    TransactionScript(TransactionScriptABI),
+    ScriptFunction(ScriptFunctionABI),
+}
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ScriptFunctionABI {
+    /// The public name of the script.
+    name: String,
+    /// The module name where the script lives
+    module_name: ModuleId,
+    /// Some text comment.
```

Comment:
Unrelated: This text comments follow markdown-ish conventions in practice. I wonder if we have documented them and/or are enforcing well formed markdown now.

Reply:
They are enforced (somewhat) in the documentation generator. But that's more of a Diem Framework/Document Generator invariant, and not one enforced by the ABI generator (yet!). I think adding this in along with some translation process in the generated documents to match the language being generated would be pretty cool